test/tester.c

```diff
@@ -3969,6 +3969,68 @@ START_TEST(test_local_disk_led_status_get)
 }
 END_TEST
 
+/*TODO(Gris Ge): Merge duplicate code of local disk test cases */
+START_TEST(test_local_disk_link_speed_get)
+{
+    int rc = LSM_ERR_OK;
+    lsm_string_list *disk_paths = NULL;
+    lsm_error *lsm_err = NULL;
+    uint32_t i = 0;
+    const char *disk_path = NULL;
+    uint32_t link_speed = LSM_DISK_LINK_SPEED_UNKNOWN;
+
+    rc = lsm_local_disk_list(&disk_paths, &lsm_err);
+    fail_unless(rc == LSM_ERR_OK, "lsm_local_disk_list() failed as %d", rc);
+    /* Only try maximum 4 disks */
+    for (; i < lsm_string_list_size(disk_paths) && i < 4; ++i) {
+        disk_path = lsm_string_list_elem_get(disk_paths, i);
+        fail_unless (disk_path != NULL, "Got NULL disk path");
+        rc = lsm_local_disk_link_speed_get(disk_path, &link_speed, &lsm_err);
+        fail_unless(rc == LSM_ERR_OK || rc == LSM_ERR_NO_SUPPORT ||
+                    rc == LSM_ERR_PERMISSION_DENIED,
+                    "lsm_local_disk_led_status_get(): "
+                    "Got unexpected return: %d", rc);
+        if (lsm_err)
+            lsm_error_free(lsm_err);
+        lsm_err = NULL;
+    }
+
+    /* Test invalid argument */
+    rc = lsm_local_disk_link_speed_get(NULL, &link_speed, &lsm_err);
+    fail_unless(rc == LSM_ERR_INVALID_ARGUMENT,
+                "Expecting LSM_ERR_INVALID_ARGUMENT, but got %d", rc);
+    fail_unless(lsm_err != NULL, "Expecting non-NULL lsm_error, but got NULL");
+    fail_unless(link_speed == LSM_DISK_LINK_SPEED_UNKNOWN,
+                "Expecting link_speed been set as LSM_DISK_LINK_SPEED_UNKNOWN, "
```

Comment:
All of these "Expecting link_speed been set..." error messages in this file should be changed to something like "Expecting link_speed to have been set...". It'll make the error messages easier to read.